docs/usage/fields.md

```diff
@@ -246,4 +281,399 @@ print(user)
+        },
+        "regex": {
+          "title": "Regex",
+          "type": "string",
+          "pattern": "^\\d*$"
+        }
+      },
+      "required": [
+        "short",
+        "long",
+        "regex"
+      ]
+    }
+    ```
+
+## Decimal Constraints
+
+There are fields that can be used to constrain decimals:
+
+* `max_digits`: Maximum number of digits within the `Decimal`. It does not include a zero before the decimal point or
+  trailing decimal zeroes.
+* `decimal_places`: Maximum number of decimal places allowed. It does not include trailing decimal zeroes.
+
+Here's an example:
+
+```py
+from decimal import Decimal
+
+from pydantic import BaseModel, Field
+
+
+class Foo(BaseModel):
+    precise: Decimal = Field(max_digits=5, decimal_places=2)
+
+
+foo = Foo(precise=Decimal('123.45'))
+print(foo)
+#> precise=Decimal('123.45')
+```
+
+## Dataclass Constraints
+
+There are fields that can be used to constrain dataclasses:
+
+* `init_var`: Whether the field should be seen as an [init-only field] in the dataclass.
+* `kw_only`: Whether the field should be a keyword-only argument in the constructor of the dataclass.
+
+Here's an example:
+
+```py
+from pydantic import BaseModel, Field
+from pydantic.dataclasses import dataclass
+
+
+@dataclass
+class Foo:
+    bar: str
+    baz: str = Field(init_var=True)
+    qux: str = Field(kw_only=True)
+
+
+class Model(BaseModel):
+    foo: Foo
+
+
+model = Model(foo=Foo('bar', baz='baz', qux='qux'))
+print(model.model_dump())  # (1)!
+#> {'foo': {'bar': 'bar', 'qux': 'qux'}}
+```
+
+1. The `baz` field is not included in the `model_dump()` output, since it is an init-only field.
+
+## Field Representation
+
+The parameter `repr` can be used to control whether the field should be included in the string
+representation of the model.
+
+```py
+from pydantic import BaseModel, Field
+
+
+class User(BaseModel):
+    name: str = Field(repr=True)  # (1)!
+    age: int = Field(repr=False)
+
+
+user = User(name='John', age=42)
+print(user)
+#> name='John'
+```
+
+1. This is the default value.
+
+## Discriminator
+
+The parameter `discriminator` can be used to control the field that will be used to discriminate between different
+models in a union.
+
+```py requires="3.8"
+from typing import Literal, Union
+
+from pydantic import BaseModel, Field
+
+
+class Cat(BaseModel):
+    pet_type: Literal['cat']
+    age: int
+
+
+class Dog(BaseModel):
+    pet_type: Literal['dog']
+    age: int
+
+
+class Model(BaseModel):
+    pet: Union[Cat, Dog] = Field(discriminator='pet_type')
+
+
+print(Model.model_validate({'pet': {'pet_type': 'cat', 'age': 12}}))  # (1)!
+#> pet=Cat(pet_type='cat', age=12)
+```
+
+1. See more about [Helper Functions] in the [Models] page.
+
+See the [Discriminated Unions] for more details.
+
+## Strict Mode
+
+The parameter `strict` can be used to control whether the field should be included in the strict mode.
+
+TODO: Should we explain what is strict mode here or add a link?
```

Comment:
If there is a section about `strict mode` add the link. otherwise a short description would be good here.

Reply:
There is a /usage/types/standard_types/#strict-types section currently, would it be a good idea to update or expand it?

Reply:
I think it makes sense to make a new section about strict mode, probably documented in the model_config section of the docs. For now let's just leave it as a TODO comment.
```suggestion
[TODO: Add a link to documentation about strict mode]
```